### PR TITLE
repo-updater: Scheduler fixes

### PR DESCRIPTION
**Pull request by tsenart**
This PR brings three improvements and/or fixes to the repo updater scheduler.

To be reviewed commit by commit.

### CODE REVIEW

File: cmd/repo-updater/repos/scheduler.go [hunk 1]
```diff
@@ -238,32 +238,20 @@ var configuredLimiter = func() *mutablelimiter.Limiter {
...
+	for _, r := range rs {
 		if r.IsDeleted() {
 			s.remove(r)
 		} else {
 			s.upsert(r)
 		}
 	}
 
-	schedKnownRepos.Set(float64(len(diff.Unmodified) + len(diff.Modified) + len(diff.Added)))
+	schedKnownRepos.Set(float64(len(rs)))
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/scheduler.go]: This metric is a gauge, so I would expect it to go down. So I think its just badly named, but the useful metric here is tracking the number of repositories that the scheduler is attempting to keep up to date. So you need to only count the upserted repos, not deleted.

**Comment by tsenart** [comment]
There's a big regression in this PR. The EnqueueRepoUpdater handler takes in a repo name, so that needs to be changed as well. E2E tests caught this by timing out.

**Comment by tsenart** [comment]
I removed the changes related to keying the scheduler data structured by ID.

**Comment by tsenart** [comment]
Related: https://github.com/sourcegraph/sourcegraph/issues/2609

**Pull request merged**